File: binding/src/charts/libqcandlestickset.cpp
#include <QBrush>
#include <QCandlestickSet>
#include <QObject>
#include <QPen>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <qcandlestickset.h>
#include "libqcandlestickset.h"
#include "libqcandlestickset.hxx"

QCandlestickSet* QCandlestickSet_new() {
	 return new QCandlestickSet();
}

QCandlestickSet* QCandlestickSet_new2(double open, double high, double low, double close) {
	 return new QCandlestickSet(open, high, low, close);
}

QCandlestickSet* QCandlestickSet_new3(double timestamp) {
	 return new QCandlestickSet(timestamp);
}

QCandlestickSet* QCandlestickSet_new4(double timestamp, QObject* parent) {
	 return new QCandlestickSet(timestamp, parent);
}

QCandlestickSet* QCandlestickSet_new5(double open, double high, double low, double close, double timestamp) {
	 return new QCandlestickSet(open, high, low, close, timestamp);
}

QCandlestickSet* QCandlestickSet_new6(double open, double high, double low, double close, double timestamp, QObject* parent) {
	 return new QCandlestickSet(open, high, low, close, timestamp, parent);
}

libqt_string QCandlestickSet_Tr(const char* s) {
	QString _ret = QCandlestickSet::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QCandlestickSet_SetTimestamp(QCandlestickSet* self, double timestamp) {
	self->setTimestamp(timestamp);
}

double QCandlestickSet_Timestamp(const QCandlestickSet* self) {
	return self->timestamp();
}

void QCandlestickSet_SetOpen(QCandlestickSet* self, double open) {
	self->setOpen(open);
}

double QCandlestickSet_Open(const QCandlestickSet* self) {
	return self->open();
}

void QCandlestickSet_SetHigh(QCandlestickSet* self, double high) {
	self->setHigh(high);
}

double QCandlestickSet_High(const QCandlestickSet* self) {
	return self->high();
}

void QCandlestickSet_SetLow(QCandlestickSet* self, double low) {
	self->setLow(low);
}

double QCandlestickSet_Low(const QCandlestickSet* self) {
	return self->low();
}

void QCandlestickSet_SetClose(QCandlestickSet* self, double close) {
	self->setClose(close);
}

double QCandlestickSet_Close(const QCandlestickSet* self) {
	return self->close();
}

void QCandlestickSet_SetBrush(QCandlestickSet* self, const QBrush* brush) {
	self->setBrush(*brush);
}

QBrush* QCandlestickSet_Brush(const QCandlestickSet* self) {
	return new QBrush(self->brush());
}

void QCandlestickSet_SetPen(QCandlestickSet* self, const QPen* pen) {
	self->setPen(*pen);
}

QPen* QCandlestickSet_Pen(const QCandlestickSet* self) {
	return new QPen(self->pen());
}

void QCandlestickSet_Clicked(QCandlestickSet* self) {
	self->clicked();
}

void QCandlestickSet_Connect_Clicked(QCandlestickSet* self, intptr_t slot) {
    void (*slotFunc)(QCandlestickSet*) = reinterpret_cast<void (*)(QCandlestickSet*)>(slot);
    QCandlestickSet::connect(self, &QCandlestickSet::clicked, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QCandlestickSet_Hovered(QCandlestickSet* self, bool status) {
	self->hovered(status);
}

void QCandlestickSet_Connect_Hovered(QCandlestickSet* self, intptr_t slot) {
    void (*slotFunc)(QCandlestickSet*, bool) = reinterpret_cast<void (*)(QCandlestickSet*, bool)>(slot);
    QCandlestickSet::connect(self, &QCandlestickSet::hovered, [self, slotFunc](bool status) {
	slotFunc(self, status);
    });
}

void QCandlestickSet_Pressed(QCandlestickSet* self) {
	self->pressed();
}

void QCandlestickSet_Connect_Pressed(QCandlestickSet* self, intptr_t slot) {
    void (*slotFunc)(QCandlestickSet*) = reinterpret_cast<void (*)(QCandlestickSet*)>(slot);
    QCandlestickSet::connect(self, &QCandlestickSet::pressed, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QCandlestickSet_Released(QCandlestickSet* self) {
	self->released();
}

void QCandlestickSet_Connect_Released(QCandlestickSet* self, intptr_t slot) {
    void (*slotFunc)(QCandlestickSet*) = reinterpret_cast<void (*)(QCandlestickSet*)>(slot);
    QCandlestickSet::connect(self, &QCandlestickSet::released, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QCandlestickSet_DoubleClicked(QCandlestickSet* self) {
	self->doubleClicked();
}

void QCandlestickSet_Connect_DoubleClicked(QCandlestickSet* self, intptr_t slot) {
    void (*slotFunc)(QCandlestickSet*) = reinterpret_cast<void (*)(QCandlestickSet*)>(slot);
    QCandlestickSet::connect(self, &QCandlestickSet::doubleClicked, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QCandlestickSet_TimestampChanged(QCandlestickSet* self) {
	self->timestampChanged();
}

void QCandlestickSet_Connect_TimestampChanged(QCandlestickSet* self, intptr_t slot) {
    void (*slotFunc)(QCandlestickSet*) = reinterpret_cast<void (*)(QCandlestickSet*)>(slot);
    QCandlestickSet::connect(self, &QCandlestickSet::timestampChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QCandlestickSet_OpenChanged(QCandlestickSet* self) {
	self->openChanged();
}

void QCandlestickSet_Connect_OpenChanged(QCandlestickSet* self, intptr_t slot) {
    void (*slotFunc)(QCandlestickSet*) = reinterpret_cast<void (*)(QCandlestickSet*)>(slot);
    QCandlestickSet::connect(self, &QCandlestickSet::openChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QCandlestickSet_HighChanged(QCandlestickSet* self) {
	self->highChanged();
}

void QCandlestickSet_Connect_HighChanged(QCandlestickSet* self, intptr_t slot) {
    void (*slotFunc)(QCandlestickSet*) = reinterpret_cast<void (*)(QCandlestickSet*)>(slot);
    QCandlestickSet::connect(self, &QCandlestickSet::highChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QCandlestickSet_LowChanged(QCandlestickSet* self) {
	self->lowChanged();
}

void QCandlestickSet_Connect_LowChanged(QCandlestickSet* self, intptr_t slot) {
    void (*slotFunc)(QCandlestickSet*) = reinterpret_cast<void (*)(QCandlestickSet*)>(slot);
    QCandlestickSet::connect(self, &QCandlestickSet::lowChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QCandlestickSet_CloseChanged(QCandlestickSet* self) {
	self->closeChanged();
}

void QCandlestickSet_Connect_CloseChanged(QCandlestickSet* self, intptr_t slot) {
    void (*slotFunc)(QCandlestickSet*) = reinterpret_cast<void (*)(QCandlestickSet*)>(slot);
    QCandlestickSet::connect(self, &QCandlestickSet::closeChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QCandlestickSet_BrushChanged(QCandlestickSet* self) {
	self->brushChanged();
}

void QCandlestickSet_Connect_BrushChanged(QCandlestickSet* self, intptr_t slot) {
    void (*slotFunc)(QCandlestickSet*) = reinterpret_cast<void (*)(QCandlestickSet*)>(slot);
    QCandlestickSet::connect(self, &QCandlestickSet::brushChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QCandlestickSet_PenChanged(QCandlestickSet* self) {
	self->penChanged();
}

void QCandlestickSet_Connect_PenChanged(QCandlestickSet* self, intptr_t slot) {
    void (*slotFunc)(QCandlestickSet*) = reinterpret_cast<void (*)(QCandlestickSet*)>(slot);
    QCandlestickSet::connect(self, &QCandlestickSet::penChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

libqt_string QCandlestickSet_Tr2(const char* s, const char* c) {
	QString _ret = QCandlestickSet::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QCandlestickSet_Tr3(const char* s, const char* c, int n) {
	QString _ret = QCandlestickSet::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QCandlestickSet_Delete(QCandlestickSet* self) {
    delete self;
}

File: binding/src/charts/libqcandlestickset.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qcandlestickset.html
pub const qcandlestickset = struct {

    /// New constructs a new QCandlestickSet object.
    pub fn New() QtC.QCandlestickSet {
        return qtc.QCandlestickSet_new();
    }


    /// New2 constructs a new QCandlestickSet object.
    pub fn New2(open: f64, high: f64, low: f64, close: f64) QtC.QCandlestickSet {
        return qtc.QCandlestickSet_new2(open, high, low, close);
    }


    /// New3 constructs a new QCandlestickSet object.
    pub fn New3(timestamp: f64) QtC.QCandlestickSet {
        return qtc.QCandlestickSet_new3(timestamp);
    }


    /// New4 constructs a new QCandlestickSet object.
    pub fn New4(timestamp: f64, parent: ?*anyopaque) QtC.QCandlestickSet {
        return qtc.QCandlestickSet_new4(timestamp, @ptrCast(parent));
    }


    /// New5 constructs a new QCandlestickSet object.
    pub fn New5(open: f64, high: f64, low: f64, close: f64, timestamp: f64) QtC.QCandlestickSet {
        return qtc.QCandlestickSet_new5(open, high, low, close, timestamp);
    }


    /// New6 constructs a new QCandlestickSet object.
    pub fn New6(open: f64, high: f64, low: f64, close: f64, timestamp: f64, parent: ?*anyopaque) QtC.QCandlestickSet {
        return qtc.QCandlestickSet_new6(open, high, low, close, timestamp, @ptrCast(parent));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QCandlestickSet_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qcandlestickset.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn SetTimestamp(self: ?*anyopaque, timestamp: f64) void {
        qtc.QCandlestickSet_SetTimestamp(@ptrCast(self), timestamp);
    }

    pub fn Timestamp(self: ?*anyopaque, ) f64 {
        return qtc.QCandlestickSet_Timestamp(@ptrCast(self));
    }

    pub fn SetOpen(self: ?*anyopaque, open: f64) void {
        qtc.QCandlestickSet_SetOpen(@ptrCast(self), open);
    }

    pub fn Open(self: ?*anyopaque, ) f64 {
        return qtc.QCandlestickSet_Open(@ptrCast(self));
    }

    pub fn SetHigh(self: ?*anyopaque, high: f64) void {
        qtc.QCandlestickSet_SetHigh(@ptrCast(self), high);
    }

    pub fn High(self: ?*anyopaque, ) f64 {
        return qtc.QCandlestickSet_High(@ptrCast(self));
    }

    pub fn SetLow(self: ?*anyopaque, low: f64) void {
        qtc.QCandlestickSet_SetLow(@ptrCast(self), low);
    }

    pub fn Low(self: ?*anyopaque, ) f64 {
        return qtc.QCandlestickSet_Low(@ptrCast(self));
    }

    pub fn SetClose(self: ?*anyopaque, close: f64) void {
        qtc.QCandlestickSet_SetClose(@ptrCast(self), close);
    }

    pub fn Close(self: ?*anyopaque, ) f64 {
        return qtc.QCandlestickSet_Close(@ptrCast(self));
    }

    pub fn SetBrush(self: ?*anyopaque, brush: ?*anyopaque) void {
        qtc.QCandlestickSet_SetBrush(@ptrCast(self), @ptrCast(brush));
    }

    pub fn Brush(self: ?*anyopaque, ) QtC.QBrush {
        return qtc.QCandlestickSet_Brush(@ptrCast(self));
    }

    pub fn SetPen(self: ?*anyopaque, pen: ?*anyopaque) void {
        qtc.QCandlestickSet_SetPen(@ptrCast(self), @ptrCast(pen));
    }

    pub fn Pen(self: ?*anyopaque, ) QtC.QPen {
        return qtc.QCandlestickSet_Pen(@ptrCast(self));
    }

    pub fn Clicked(self: ?*anyopaque, ) void {
        qtc.QCandlestickSet_Clicked(@ptrCast(self));
    }

    pub fn OnClicked(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QCandlestickSet_Connect_Clicked(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Hovered(self: ?*anyopaque, status: bool) void {
        qtc.QCandlestickSet_Hovered(@ptrCast(self), status);
    }

    pub fn OnHovered(self: ?*anyopaque, callback: *const fn (?*anyopaque, bool) callconv(.c) void) void {
        qtc.QCandlestickSet_Connect_Hovered(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Pressed(self: ?*anyopaque, ) void {
        qtc.QCandlestickSet_Pressed(@ptrCast(self));
    }

    pub fn OnPressed(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QCandlestickSet_Connect_Pressed(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Released(self: ?*anyopaque, ) void {
        qtc.QCandlestickSet_Released(@ptrCast(self));
    }

    pub fn OnReleased(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QCandlestickSet_Connect_Released(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn DoubleClicked(self: ?*anyopaque, ) void {
        qtc.QCandlestickSet_DoubleClicked(@ptrCast(self));
    }

    pub fn OnDoubleClicked(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QCandlestickSet_Connect_DoubleClicked(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn TimestampChanged(self: ?*anyopaque, ) void {
        qtc.QCandlestickSet_TimestampChanged(@ptrCast(self));
    }

    pub fn OnTimestampChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QCandlestickSet_Connect_TimestampChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn OpenChanged(self: ?*anyopaque, ) void {
        qtc.QCandlestickSet_OpenChanged(@ptrCast(self));
    }

    pub fn OnOpenChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QCandlestickSet_Connect_OpenChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn HighChanged(self: ?*anyopaque, ) void {
        qtc.QCandlestickSet_HighChanged(@ptrCast(self));
    }

    pub fn OnHighChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QCandlestickSet_Connect_HighChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn LowChanged(self: ?*anyopaque, ) void {
        qtc.QCandlestickSet_LowChanged(@ptrCast(self));
    }

    pub fn OnLowChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QCandlestickSet_Connect_LowChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn CloseChanged(self: ?*anyopaque, ) void {
        qtc.QCandlestickSet_CloseChanged(@ptrCast(self));
    }

    pub fn OnCloseChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QCandlestickSet_Connect_CloseChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn BrushChanged(self: ?*anyopaque, ) void {
        qtc.QCandlestickSet_BrushChanged(@ptrCast(self));
    }

    pub fn OnBrushChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QCandlestickSet_Connect_BrushChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn PenChanged(self: ?*anyopaque, ) void {
        qtc.QCandlestickSet_PenChanged(@ptrCast(self));
    }

    pub fn OnPenChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QCandlestickSet_Connect_PenChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QCandlestickSet_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qcandlestickset.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QCandlestickSet_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qcandlestickset.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QCandlestickSet_Delete(@ptrCast(self));
    }
};
